src/librustc_codegen_llvm/builder.rs

```diff
@@ -564,6 +564,14 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
 
 
     pub fn range_metadata(&self, load: ValueRef, range: Range<u128>) {
+        if self.sess().target.target.arch == "amdgpu" {
+            // amdgpu/LLVM does something weird and thinks a i64 value is
```

Comment:
This sounds like a bug in LLVM. Consider filling a (rust-lang) issue and referencing this piece of code there.

It is definitely desirable to maintain range metadata somehow.